hibernate-reactive-core/src/test/java/org/hibernate/reactive/ReactiveStatelessUpdateTest.java

```diff
@@ -0,0 +1,147 @@
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+
+import org.hibernate.cfg.Configuration;
+
+import org.junit.After;
+import org.junit.Test;
+
+import io.vertx.ext.unit.TestContext;
+
+public class ReactiveStatelessUpdateTest  extends BaseReactiveTest {
+
+	@Override
+	protected Configuration constructConfiguration() {
+		Configuration configuration = super.constructConfiguration();
+		configuration.addAnnotatedClass( SampleEntity.class );
+		configuration.addAnnotatedClass( SampleJoinEntity.class );
+		return configuration;
+	}
+
+	@After
+	public void cleanDb(TestContext context) {
+		test( context, deleteEntities( "SampleEntity", "SampleJoinEntity" ) );
+	}
+
+	@Test
+	public void testUpdateWithDetachedEntity(TestContext context) {
+		SampleEntity sampleEntity = new SampleEntity();
+		sampleEntity.sampleField = "test";
+
+		SampleJoinEntity sampleJoinEntity = new SampleJoinEntity();
+		sampleJoinEntity.sampleEntity = sampleEntity;
+
+		test( context, getMutinySessionFactory()
+				.withTransaction( (s, t) -> s.persistAll( sampleEntity, sampleJoinEntity ) )
+
+				.chain( targetId -> getMutinySessionFactory()
+						.withTransaction( (session, transaction) -> session
+								.find( SampleJoinEntity.class, sampleJoinEntity.id ) )
+				)
+
+				.chain( joinEntityFromDatabase -> getMutinySessionFactory().withStatelessTransaction(
+						(s, t) -> {
+							SampleEntity entityFromDb = joinEntityFromDatabase.sampleEntity;
+							entityFromDb.sampleField = "updated field";
+							// Intended to test updating unmanaged proxy entity in this transaction
+							// @See ReactiveStatelessSessionImpl.updateReactive()
+							return s.update( entityFromDb );
+						} )
+				)
+				.invoke( () -> getMutinySessionFactory().withSession( session -> session
+						.createQuery( "from SampleEntity", SampleEntity.class )
+						.getSingleResult()
+						.onItem().invoke( () -> context.assertEquals("updated field", sampleEntity.sampleField  ) ))
+				)
+		);
+	}
+
+	@Test
+	public void testUpdateWithManagedProxyEntity(TestContext context) {
+		SampleEntity sampleEntity = new SampleEntity();
+		sampleEntity.sampleField = "test";
+
+		SampleJoinEntity sampleJoinEntity = new SampleJoinEntity();
+		sampleJoinEntity.sampleEntity = sampleEntity;
+
+		test( context, getMutinySessionFactory()
+				.withTransaction( (s, t) -> s.persistAll( sampleEntity, sampleJoinEntity ) )
+
+				.chain( targetId -> getMutinySessionFactory()
+						.withTransaction( (session, transaction) -> session
+								.find( SampleJoinEntity.class, sampleJoinEntity.id ) )
+				)
+
+				.chain( joinEntityFromDatabase -> getMutinySessionFactory().withStatelessTransaction(
+						(s, t) -> s
+								// fetching entity in this transaction to trigger logic for
+								// updating a managed proxy entity
+								// @See ReactiveStatelessSessionImpl.updateReactive()
```

Comment:
I would add this comment on the javadoc for the class, so that it's actually linkable:
```
/**
 * @see org.hibernate.reactive.session.impl.ReactiveStatelessSessionImpl#reactiveUpdate(Object)
 */
public class ReactiveStatelessUpdateTest  extends BaseReactiveTest {
```